File: src/main/scala/com/gprasad/hackerrank/scalasol/algo/warmup/SimpleArraySum.scala
package com.gprasad.hackerrank.scalasol.algo.warmup

/**
  * @author gq6pras
  */
object SimpleArraySum {
  def main(args: Array[String]) {
    val sc = new java.util.Scanner(System.in);
    var n = sc.nextInt();
    var arr = new Array[Int](n);
    for (arr_i <- 0 to n - 1) {
      arr(arr_i) = sc.nextInt();
    }
    //arr.sum
    /*var i = 0
    var sum = 0
    while (i < n) {
      sum += arr(i)
      i += 1
    }*/
    val result = simpleArraySum(n, arr)
    println(result)
  }

  def simpleArraySum(n: Int, ar: Array[Int]): Int =  {
    // Complete this function
    ar.sum
  }
}

File: src/main/java/com/gprasad/hackerrank/javasol/algo/warmup/SimpleArraySum.java
/*
 

Problem Statement

You are given an array of integers of size N. Can you find the sum of the elements in the array?

Input
The first line of input consists of an integer N. The next line contains N space-separated integers representing the array elements.
Sample:

6

1 2 3 4 10 11

Output
Output a single value equal to the sum of the elements in the array.
For the sample above you would just print 31 since 1+2+3+4+10+11=31.

 */
package com.gprasad.hackerrank.javasol.algo.warmup;

import java.util.Scanner;

/**
 *
 * @author gq6pras
 */
public class SimpleArraySum {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int arrSize = input.nextInt();
        int[] intArr = new int[arrSize];
        for(int i =0 ; i<arrSize; i++)
        {
            intArr[i] = input.nextInt();
        }
        int sum = 0;
        for(Integer num : intArr)
        {
            sum = sum+num;
        }
        System.out.println(sum);
    }
}
